sys/parsoid_proxy.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+/*
+ * Simple proxy to route requests to the client-requested
+ * Parsoid variant (JS or PHP) during the transition
+ * period. Parsoid/JS is being phased out and replaced by
+ * Parsoid/PHP.
+ */
+
+const HyperSwitch = require('hyperswitch');
+const URI = HyperSwitch.URI;
+const mwUtil = require('../lib/mwUtil');
+const VARIANT_HDR_NAME = 'x-parsoid-variant';
+
+module.exports = () => {
+    return {
+        spec: {
+            paths: {
+                '/{+path}': {
+                    all: {
+                        operationId: 'proxy_parsoid_variant'
+                    }
+                }
+            }
+        },
+        operations: {
+            proxy_parsoid_variant: (hyper, req) => {
+                const rootReqHeaders = hyper._rootReq.headers || {};
+                rootReqHeaders[VARIANT_HDR_NAME] = rootReqHeaders[VARIANT_HDR_NAME] || 'JS';
+                const isPhpVariant = /PHP/i.test(rootReqHeaders[VARIANT_HDR_NAME]);
+
+                return hyper.request({
+                    method: req.method,
+                    uri: new URI(req.uri.toString().replace('/parsoid/',
+                        isPhpVariant ? '/parsoidphp/' : '/parsoidjs/')),
+                    headers: req.headers,
+                    body: req.body,
+                    query: req.query,
+                    params: req.params
+                })
+                .then((res) => {
```

Comment:
1. can be replaced with `tap`
2. if 'res' is falsy, we're in trouble... Maybe no need to check for it to be truthy

Reply:
1. Replace with tap. 2. Removed